src/machine/system_calls.rs

```diff
@@ -104,6 +104,13 @@ use super::libraries;
 use super::preprocessor::to_op_decl;
 use super::preprocessor::to_op_decl_spec;
 
+/// Represents the presence (or absence) of a `module:` prefix to predicates.
+///
+/// On the prolog side, `strip_module(X, Y, Z)` takes care of splitting the `X = module:predicate`
+/// pair into `Y = module` and `Z = predicate`. If no module prefix is present, then `Y` is left unset.
```

Comment:
Is this comment accurate, specifically "then `Y` is left unset"? It seems `Y` is either `specified(M)` or **`unspecified`**, at least these are the cases that `strip_module/3` above distinguishes.

Every comment carries the risk of introducing wrong statements that cannot be verified automatically.

Reply:
You're right, that comment was left out of my last round of edits